src/server/pfs/server/driver_v2.go

```diff
@@ -168,39 +165,87 @@ func (d *driverV2) withFileSet(ctx context.Context, repo, commit string, f func(
 	})
 }
 
-// TODO Need to figure out path cleaning.
-func (d *driverV2) getTar(ctx context.Context, repo, commit, glob string, w io.Writer) error {
-	if err := d.getTarConditional(ctx, repo, commit, glob, func(fr *FileReader) error {
-		return fr.Get(w, true)
-	}); err != nil {
+func (d *driverV2) getTar(ctx context.Context, commit *pfs.Commit, glob string, w io.Writer) error {
+	compactedPaths := []string{compactedCommitPath(commit)}
+	prefix := globLiteralPrefix(glob)
+	mf, err := matchFunc(glob)
+	if err != nil {
 		return err
 	}
-	// Close a tar writer to create tar EOF padding.
-	return tar.NewWriter(w).Close()
+	mr, err := d.storage.NewMergeReader(ctx, compactedPaths, index.WithPrefix(prefix))
+	if err != nil {
+		return err
+	}
+	filter := &fileset.IndexFilter{
+		R: mr,
+		F: func(idx *index.Index) bool {
+			return mf(idx.Path)
+		},
+	}
+	// TODO: remove absolute paths on the way out?
+	// nonAbsolute := &fileset.HeaderMapper{
+	// 	R: filter,
+	// 	F: func(th *tar.Header) *tar.Header {
+	// 		th.Name = "." + th.Name
+	// 		return th
+	// 	},
+	// }
+	return fileset.WriteTarStream(w, filter)
 }
 
-func (d *driverV2) listFileV2(pachClient *client.APIClient, file *pfs.File, full bool, history int64, f func(*pfs.FileInfoV2) error) error {
+func (d *driverV2) listFileV2(pachClient *client.APIClient, file *pfs.File, full bool, history int64, cb func(*pfs.FileInfoV2) error) error {
```

Comment:
Few minor nits for this function:
- In general, if you want to add comments, then I think they need to add more information than can be gleaned from the code (variable / function names). The comments in this function don't seem like they add any value and they take up space.
- `finfo` -> `fi`. Similarly to `cb`, having a common naming convention for even simple local variable names really adds up in terms of removing mental overhead when reading code.
- When the last statement in a function that returns an error is just a function call that returns an error, then you should just return the result of the function call. It is a pretty trivial detail, but I am convinced that simplifying wherever you can really adds up.